File: java/mission-to-mars/src/test/java/mars/SondeTest.java
package mars;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class SondeTest {

	@Test
	void roverDoitEtreRecupere() {
		// Arrange
		Sonde sut = new Sonde();
		ModuleMartien perseverance = new Rover(Direction.NORD, new Position(4, 4, 0));

		// Act
		sut.preparerRecuperation(perseverance);

		// Assert
		assertTrue(perseverance.isPretPourRecuperation());
		assertEquals(new Position(4, 4, 0), perseverance.getPosition());
	}

	@Test
	void helicoptereDoitEtreRecupere() {
		// Arrange
		Sonde sut = new Sonde();
		ModuleMartien ingenuity = new Helicoptere(Direction.NORD, new Position(4, 4, 50));

		// Act
		sut.preparerRecuperation(ingenuity);

		// Assert
		assertTrue(ingenuity.isPretPourRecuperation());
		assertEquals(new Position(4, 4, 0), ingenuity.getPosition());
	}
}

File: typescript/tests/Sonde.test.ts
import { Direction } from '../src/Direction'
import { Helicoptere } from '../src/Helicoptere'
import { ModuleMartien } from '../src/ModuleMartien'
import { Position } from '../src/Position'
import { Rover } from '../src/Rover'
import { Sonde } from '../src/Sonde'

describe('Sonde', () => {
  it('Le rover doit être récupéré', () => {
    // GIVEN
    const perseverance: ModuleMartien = new Rover(Direction.NORD, new Position(4, 4, 0));
    const sonde: Sonde = new Sonde();

    // WHEN
    sonde.preparerRecuperation(perseverance);

    // THEN
    expect(perseverance.isPretPourRecuperation()).toBeTruthy();
    expect(new Position(4, 4, 0)).toStrictEqual(perseverance.getPosition())
  })

  it('L’hélicoptère doit être récupéré', () => {
    // GIVEN
    const ingenuity: ModuleMartien = new Helicoptere(Direction.NORD, new Position(4, 4, 50));
    const sonde: Sonde = new Sonde();

    // WHEN
    sonde.preparerRecuperation(ingenuity);

    // THEN
    expect(ingenuity.isPretPourRecuperation()).toBeTruthy();
    expect(new Position(4, 4, 0)).toStrictEqual(ingenuity.getPosition())
  })
})
